content/articles/lru-cache-implementation-in-javascript/index.md

```diff
@@ -0,0 +1,135 @@
+            var LRUCache = function(capacity) {
+                this.capacity = capacity;
+            this.map = new Map(); //this stores the entire array
+
+            //this is boundaries for double Linked List
+            this.head = {};
+            this.tail = {};
+            this.head.next = this.tail; //initialize your double Linked List
+            this.tail.prev = this.head;
+            };
+
+            ```
+
+2.  Get operation. This will return the value of the Key if it exists. Else, it returns -1.
+
+            ````js
+
+            LRUCache.prototype.get = function(key) {
+            if(this.map.has(key)){
+            //remove elem from current position
+            let c = this.map.get(key);
+            c.prev.next = c.next;
+            c.next.prev = c.prev;
+
+            this.tail.prev.next = c; //insert it after last element. Element before tail
+            c.prev = this.tail.prev; //update c.prev and next pointer
+            c.next = this.tail;
+            this.tail.prev = c; //update last element as tail
+            return c.value;
+            } else {
+            return -1; //element does not exist
+            }
+
+            };
+
+    ```
+
+    ```
+
+3.  Put operation. The put operation will update the value of the key if it is found. If found, add the Key and the value pair to the cache. If the number of keys has exceeded the initialized capacity of the cache, evict the least recently accessed item.
+
+           ```js
+           LRUCache.prototype.put = function(key, value) {
+
+            if(this.get(key) !== -1){ //key does not exist, update last element value
+            this.tail.prev.value = value;
+            } else {
+            //need to check if map size is at capacity
+            if(this.map.size === this.capacity) {
+            //delete item both from map and DLL
+            this.map.delete(this.head.next.key); //delete first element of list
+            this.head.next = this.head.next.next; //update first element as next element
+            this.head.next.prev = this.head;
+            }
+
+            let newNode = {
+            value,
+            key
+            }; //each node is a hashtable that stores key and value
+
+            //When adding a new node, we need to update both map and DLL
+            this.map.set(key, newNode); //add current node to map
+            this.tail.prev.next = newNode; //add node to end of the list
+            newNode.prev = this.tail.prev; //update prev and next pointers of newNode
+            newNode.next = this.tail;
+            this.tail.prev = newNode; //update last element
+            }
+
+            };
+            ```
+
+### Conclusion
+
+The best data structure to use when implementing LRU cache is Double Linked List and Hash Table. The time complexity for this implementation is O(1). You can go ahead and try it out on your own as well.
+
+Happy Coding! :-)
+
+### References
+
+https://leetcode.com/problems/lru-cache/

```

Comment:
Use the correct syntax highlighting for markdown links.

```md
[LRU Cache - LeetCode](https://leetcode.com/problems/lru-cache/)
```

Reply:
I think this should be it having counter-checked with the documentation. An example I got from the documentation is :-
My favorite search engine is [Duck Duck Go](https://duckduckgo.com).

Reply:
[LRU Cache - LeetCode](https://leetcode.com/problems/lru-cache/)

Reply:
I think this should be it having counter checked with the documentation. An example I got from the documentation is 
My favorite search engine is [Duck Duck Go](https://duckduckgo.com).

Reply:
@erixbiz The suggested syntax is meant to give your link a description.

Reply:
![Capture](https://user-images.githubusercontent.com/84809276/133430256-b02a5606-c94e-4ecc-ab81-13a13319d95f.PNG)
Sure. It shows that when I preview this file

Reply:
I have identified the problem and corrected. Thanks